velox/functions/prestosql/geospatial/GeometryUtils.cpp

```diff
@@ -23,6 +23,61 @@ using geos::operation::valid::IsValidOp;
 
 namespace facebook::velox::functions::geospatial {
 
+GeometryCollectionIterator::GeometryCollectionIterator(
+    const geos::geom::Geometry* geometry) {
+  if (!geometry) {
+    throw std::invalid_argument("geometry is null");
+  }
+  geometriesDeque.push_back(geometry);
+}
+// Returns true if there is a next geometry to iterate over
+bool GeometryCollectionIterator::hasNext() {
+  while (!geometriesDeque.empty()) {
+    const geos::geom::Geometry* top = geometriesDeque.back();
+    // Check if top is a GeometryCollection
+    if (top->getGeometryTypeId() == geos::geom::GEOS_GEOMETRYCOLLECTION) {
+      geometriesDeque.pop_back();
+      const geos::geom::GeometryCollection* collection =
+          dynamic_cast<const geos::geom::GeometryCollection*>(top);
+      if (collection) {
+        // Push children in reverse order so that the first child is on top of
+        // the stack
+        for (int i = static_cast<int>(collection->getNumGeometries()) - 1;
```

Comment:
nit: use auto i = .. (so you dont have to static_cast).

Reply:
We need to cast to a signed numeric type since `auto` will set the type and `getNumGeometries` returns an unsigned type. We could use auto, but we'd need to put the loop in a conditional where `collection->getNumGeometries != 0`, so I went with this since it felt a little cleaner.